examples/dev_flow/debug/step_into_dependencies.rst

```diff
@@ -0,0 +1,84 @@
+
+.. _examples_dev_flow_debug_step_into:
+
+
+Debugging and stepping into dependencies 
+========================================
+
+Sometimes, when developing and debugging your own code, it could be useful to be able to step-into the
+dependencies source code too. There are a couple of things to take into account:
+
+- Recipes and packages from ConanCenter do not package always all the debug artifacts necessary to debug. For example in Windows, the ``*.pdb`` files are not packaged, because they are very heavy, and in practice barely used. It is possible to have your own packages to package the PDB files if you want, but that still won't solve the next point.
+- Debug artifacts are often not relocatable, that means that such artifacts can only be used in the location they were built from sources. But packages that are uploaded to a server and downloaded to a different machine can put those artifacts in a different folder. Then, the debug artifacts might not correctly locate the source code, the symbols, etc.
+
+
+Building from source
+--------------------
+
+The recommended approach for debugging dependencies is building them from source in the local cache. This approach should work out of the box for most recipes, including ConanCenter recipes. 
+
+We can reuse the code from the very first example in the tutorial for this use case. Please, first clone the sources to recreate this project, you can find them in the
+`examples2.0 repository <https://github.com/conan-io/examples2>`_ in GitHub:
+
+.. code-block:: bash
+
+    $ git clone https://github.com/conan-io/examples2.git
+    $ cd examples2/tutorial/consuming_packages/simple_cmake_project
+
+Then, lets make sure the dependency is built from source:
+
+.. code-block:: bash
+
+    $ conan install . -s build_type=Debug --build="zlib/*"
+    ...
+    Install finished successfully
+
+Assuming that we have CMake>=3.23, we can use the presets (otherwise, please use the ``-DCMAKE_TOOLCHAIN_FILE`` arguments):
```

Comment:
are these some pre-defined presets or where are they coming from?

Reply:
Created by Conan CMakeToolchain, they are already described and used in other places in the docs.